src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,64 @@
+package duke.command;
+
+import duke.storage.Storage;
+import duke.task.Task;
+import duke.tasklist.TaskList;
+import duke.ui.Ui;
+
+import java.io.IOException;
+
+
+/**
+ * General command class,
+ * contains basic abstract command actions, such as execute, check if exit command
+ * and get command type and pending tasks of command.
+ */
+public abstract class Command {
+	private CommandType commandType;
+	private Task pendingTask;
+	
+	/**
+	 * Constructor for class Duke.
+	 *
+	 * @param command     String command input from user.
+	 * @param pendingTask Task object from information given by user.
+	 */
+	public Command(String command, Task pendingTask) {
```

Comment:
it seems like not all commands pertain to a particular task (e.g. bye, help), so would the code be more self-explanatory if pendingTask is removed from the method signature?